### PR TITLE
Split config into default (current) and base (without a react dependency)

### PR DESCRIPTION
**Pull request by AaronMoat**
The motivation would be to be able to update [eslint-config-skuba](https://github.com/seek-oss/eslint-config-skuba), being used for backends, to depend on the base and to get rid of the warning "Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting.".

### CODE REVIEW

File: .changeset/sixty-pants-explain.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"eslint-config-seek": patch
+---
+
+feat: Split config into default (current) and base (without a react dependency)
```
Comment (mrm007) [inline .changeset/sixty-pants-explain.md:5]:
````suggestion
Split config into default (current) and base (without a React dependency).

Without React support:

```json
{
  "extends": "seek/base"
}
```

With React support:

```json
{
  "extends": "seek"
}
```
````

File: package.json [hunk 2]
```diff
@@ -2,9 +2,10 @@
   "name": "eslint-config-seek",
   "version": "10.2.0",
   "description": "ESLint configuration used by SEEK",
-  "main": ".eslintrc.js",
+  "main": "index.js",
   "files": [
-    ".eslintrc.js"
```
Comment (mrm007) [inline package.json:7]: I would also remove `.eslintrc.js` from the repo, and keep only `index.js` and `base.js`.

File: package.json [hunk 3]
```diff
@@ -17,7 +18,7 @@
   },
   "scripts": {
     "release": "changeset publish",
-    "test": "eslint .",
+    "test": "eslint . && eslint -c base.js --no-eslintrc .",
```
Comment (mrm007) [inline package.json:21]:
```suggestion
    "test": "eslint . && eslint --config base.js --no-eslintrc .",
```

Comment (AaronMoat) [reply package.json]: I've removed .eslintrc.js and made the test command explicit about which file it's using to test.

File: .changeset/sixty-pants-explain.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
+"eslint-config-seek": patch
```
Comment (mrm007) [inline .changeset/sixty-pants-explain.md:2]: Sorry, missed one small detail:.

```suggestion
"eslint-config-seek": minor
```

**Review by mrm007 — APPROVED** [review]

**Pull request merged**